### ISSUE TITLE
Social.localUser.Authenticate fails when using Unity Cloud Build

### ISSUE THREAD

**Issue by leonidumanskiy** [playgameservices/play-games-plugin-for-unity#2555]
I have integrated this plugin and after a bunch of tests with local builds and Unity Cloud Build, it seems that APK built with Unity Cloud Build fails to authenticate.

1. After installing locally-build APK using adb install, `Social.localUser.Authenticate` succeeds and small "welcome" banner with my name appears like this one:

![1](https://user-images.githubusercontent.com/7004323/56947227-53f10100-6ae1-11e9-8a9f-e0a5e61ccd8a.png)

2. After installing apk produced by Unity Cloud Build, "Google Play Games" popup Window will appear. After that, `Social.localUser.Authenticate` callback returns "false":

![12](https://user-images.githubusercontent.com/7004323/56947580-73d4f480-6ae2-11e9-801e-a0c843dfdfe9.png)

Here is my log (reduced for easier reading):
```
04-30 01:09:43.684 7140-7208/com.luzaro.zpgclient W/Unity: !!! [Play Games Plugin DLL] 04/30/19 1:09:43 -07:00 WARNING: Creating new PlayGamesPlatform
04-30 01:09:43.685 7140-7208/com.luzaro.zpgclient I/Unity:  [Play Games Plugin DLL] 04/30/19 1:09:43 -07:00 DEBUG: Activating PlayGamesPlatform.
     [Play Games Plugin DLL] 04/30/19 1:09:43 -07:00 DEBUG: PlayGamesPlatform activated: GooglePlayGames.PlayGamesPlatform
04-30 01:09:43.686 7140-7208/com.luzaro.zpgclient I/Unity:  [Play Games Plugin DLL] 04/30/19 1:09:43 -07:00 DEBUG: Creating platform-specific Play Games client.
04-30 01:09:43.687 7140-7208/com.luzaro.zpgclient I/Unity:  [Play Games Plugin DLL] 04/30/19 1:09:43 -07:00 DEBUG: Creating Android IPlayGamesClient Client
04-30 01:09:44.444 7140-7208/com.luzaro.zpgclient D/Unity: Sensor :        Accelerometer ( 1) ; 0.002396 / 0.00s ; LSM6DSM Accelerometer / STMicroelectronics 
04-30 01:09:44.450 7140-7140/com.luzaro.zpgclient I/Unity: windowFocusChanged: false
04-30 01:09:49.687 7140-7140/com.luzaro.zpgclient D/GamesUnitySDK: Performing Android initialization of the GPG SDK
04-30 01:09:49.699 7140-7140/com.luzaro.zpgclient I/Unity: Building GPG services, implicitly attempts silent auth
04-30 01:09:49.939 7140-7140/com.luzaro.zpgclient I/Unity: onResume
04-30 01:09:49.940 7140-7140/com.luzaro.zpgclient I/Unity: windowFocusChanged: true
04-30 01:09:49.946 7140-7208/com.luzaro.zpgclient D/Unity: Sensor :        Accelerometer ( 1) ; 0.002396 / 0.00s ; LSM6DSM Accelerometer / STMicroelectronics 
04-30 01:09:49.957 7140-7208/com.luzaro.zpgclient D/Unity: Choreographer available: Enabling VSYNC timing

V ------------- THIS ---------------
04-30 01:09:50.022 7140-7208/com.luzaro.zpgclient E/Unity: GooglePlayGamesException: Social.localUser.Authenticate return false
04-30 01:09:50.582 7140-7208/com.luzaro.zpgclient I/Unity:  [Play Games Plugin DLL] 04/30/19 1:09:50 -07:00 DEBUG: Invoking user callback on game thread
04-30 01:09:50.584 7140-7208/com.luzaro.zpgclient I/Unity:  [Play Games Plugin DLL] 04/30/19 1:09:50 -07:00 DEBUG: Starting Auth Transition. Op: SIGN_IN status: ERROR_NOT_AUTHORIZED
04-30 01:09:50.585 7140-7208/com.luzaro.zpgclient I/Unity:  [Play Games Plugin DLL] 04/30/19 1:09:50 -07:00 DEBUG: AuthState == Unauthenticated calling auth callbacks with failure
```

2.1. Sometimes, this popup will also display a list with active accounts (I only have one account, which I have registered in a list of testers in my GPG app).
2.2. I was able to successfully authenticate with this popup once, after I launched my game through Google Play Games android app. However this only happened once and I was unable to replicate it again.

**However, there is a catch.** 
Even when I uninstall one APK and install another one, authentication result seems to be cached until phone is restarted .
For instance, if I install locally built apk, authenticate and then install Cloud Build apk, it seems to **work until I restart my phone**. 
And vice versa - if I install Cloud Build apk, fail to authenticate, and then install local build, it **seems to fail until my phone is restarted**.

Any ideas?

**Comment by olehkuznetsov** [comment]
It seems if you are using Unity Cloud Build it signs the apk with different SHA1 key, so you need to link the same package name with another SHA1 key in Google Play Console.

**Comment by leonidumanskiy** [comment]
Hi Oleh,
For both builds, I am using my debug.keystore, which I also uploaded and selected for signing in Unity Cloud Build. 

The only thing that I noticed now is it says "Debug: no" in Unity cloud Build when select my keystore. I am unsure why.

**Comment by olehkuznetsov** [comment]
Can you, please, check SHA1 key for both apk generated like mentioned here?
https://stackoverflow.com/questions/11331469/how-do-i-find-out-which-keystore-was-used-to-sign-an-app

**Comment by leonidumanskiy** [comment]
@olehkuznetsov 
I compared two apk and it look like APK built by Unity Cloud Built does not have .RSA file, it has .DSA (apk signed using SHA1withDSA) file instead with a completely different SHA1.

At this point, I am very confused because I have only one debug.keystore with one entry (androiddebugkey) which I uploaded to cloud build and entered my alias, password etc.

I don't see any other resolution rather to post to Unity cloud build forums. Thank you for your support.
